### DISCUSSION TITLE
Moes Tuya Zigbee rotary dimmer switch - change dimming levels

### DISCUSSION THREAD

**Discussion by woody166** [zigpy/zha-device-handlers#2089] [Q&A]
Hi All,

I have just installed the Moes Tuya Zigbee rotary dimmer switch, but the issue I'm having is that it doesn't really dim the bulb as much as I would like.  The controls for the switch in ZHA shows the brightness down to 1 but the bulb is still fairly bright, I would expect the bulb to be almost off/very dim but it like at 50% of it's original brightness.

I have installed the latest custom quirk to see if that helps but I don't see any difference.

Is there a way to alter this so the bulb can become very dim/almost off? 

Any help/comments/suggestions would be very welcome.

**Comment by MattWestb** [comment]
The ROK is only sending light level step commands and is being larger steps if rotating it faster.
So is you is getting up and the light is at 100% its cant being brighter and if you is rotating down and the light is at 1% it cant being darker.
I think you cant getting it darker by sending commands from ZHA then the remote is doing all its can going to the end if the light level limits.
You can trying one other light and see how its working then different lights have different implementations how they is doing the deeming.

The last ZHA quirk shall being updated also with tuya magic for all TS004F device so no need installing local quirks for this devices and can making problem later if ZHA is changing things.
If your device is having one new device ID pleas post it and i adding it in the quirk in ZHA.

**Comment by Hedda** [comment]
@woody166 Start by opening a new issue with "Device support request" -> https://github.com/zigpy/zha-device-handlers/issues

There you provide the requested device signature, diagnostic information, logs, etc. which makes it easier for others to answer and help.

PS: This is in ZHA docs -> https://www.home-assistant.io/integrations/zha#how-to-add-support-for-new-and-unsupported-devices
